packages/xforms-engine/src/instance/children.ts

```diff
@@ -68,6 +65,10 @@ const isSelectDefinition = (definition: ControlNodeDefinition): definition is Se
 	return definition.bodyElement.type === 'select' || definition.bodyElement.type === 'select1';
 };
 
+const isRankDefinition = (definition: ControlNodeDefinition): definition is RankDefinition => {
+	return definition.bodyElement.type === 'rank';
```

Comment:
In another PR, we can define all these strings (`rank`, `select`, etc) in one place in the engine. 
```
enum NodeType {
  Rank = "rank",
  Select = "select",
  Note = "note",
}
```

Reply:
So... let's talk more about enums! I personally like them (well, string enums like this here). But they're generally frowned upon in TS, and likely to be either deprecated or unsupported in future TS updates and/or other target environments.

Currently, we only use them in two places and I was kind of fighting myself over that when I added them.

Reply:
In that case, we could use regular objects or maps.

```
// Option 1
const NodeType {
  Rank: "rank",
  Select: "select",
  Note: "note",
}
Object.freeze(NodeType); 

// Option 2
const NodeType = new Map([
  ["Rank", "rank"],
  ["Select", "select"],
  ["Note", "note"]
]);
```

Reply:
Tracked in #296!